src/lib.rs

```diff
@@ -414,6 +414,31 @@ fn test_bit_iterator_length() {
 pub use self::arith_impl::*;
 
 mod arith_impl {
+    #[cfg(target_arch = "wasm32")]
+    #[inline(always)]
+    pub fn mul_u64(x: u64, y: u64) -> u128 {
+        // x = x_1 * 2^32 + x_0
+        let x_0 = (x as u32) as u64;
+        let x_1 = x >> 32;
+
+        // y = y_1 * 2^32 + y_0
+        let y_0 = (y as u32) as u64;
+        let y_1 = y >> 32;
+
+        let z_0 = x_0 * y_0;
+        let z_2 = x_1 * y_1;
+
+        let z_1 = u128::from(x_0 * y_1) + u128::from(x_1 * y_0);
+
+        (u128::from(z_2) << 64) + (z_1 << 32) + u128::from(z_0)
```

Comment:
Is there is any difference between using `u128::from(x)` and `x as u128`? Does it impact the performance in any way?

Reply:
`u128::from` internally uses `as` conversion:
https://doc.rust-lang.org/src/core/convert/num.rs.html#109
https://doc.rust-lang.org/src/core/convert/num.rs.html#53
So, should be the same in terms of performance. I guess, the only difference is that `as` allows for lossy conversions from larger types to smaller ones, but in this case it doesn't matter.

Reply:
Yes, I've tried both methods but it looks like they have the same performance.